### PR TITLE
shared node value consistency

### PR DESCRIPTION
**Pull request by PhilipDeegan**

### CODE REVIEW

**Comment by nicolasaunai** [comment]
see https://github.com/LLNL/SAMRAI/issues/170 updates in SAMRAI parts...

File: tests/simulator/test_advance.py [hunk 1]
```diff
@@ -235,30 +236,28 @@ def test_overlaped_fields_are_equal(self, refinement_boxes):
...
-        # {"L0": [Box3D(10, 19)]},
     )
     def test_overlaped_fields_are_equal_with_min_max_patch_size_of_max_ghosts(self, refinement_boxes):
         from pyphare.pharein.simulation import check_patch_size
 
-        dim = refinement_boxes["L0"][0].ndim
+        dim = 1
 
-        cells = [30] * dim
+        cells = [60] * dim
```
Comment (PhilipDeegan) [inline tests/simulator/test_advance.py:247]: track this with SAMRAI issue about patch sizes that are too small (<6)

Comment (PhilipDeegan) [reply tests/simulator/test_advance.py:247]: https://github.com/PHAREHUB/PHARE/issues/577

File: src/amr/messengers/ghost_only_variable_fill_pattern.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        bool overwrite_interior = true; // replace func param
+        assert(overwrite_interior_ == overwrite_interior);
+
+        auto& dst_cast = dynamic_cast<AFieldGeometry const&>(dst_geometry);
+        auto& src_cast = dynamic_cast<AFieldGeometry const&>(src_geometry);
+
+        // for shared border node value sync
+        if (src_cast.patchBox.getGlobalId().getOwnerRank()
+            != dst_cast.patchBox.getGlobalId().getOwnerRank())
+            overwrite_interior = src_cast.patchBox.getLocalId() > dst_cast.patchBox.getLocalId();
```
Comment (nicolasaunai) [inline src/amr/messengers/ghost_only_variable_fill_pattern.h:38]: GlobalID instead? 2#0 > 3#1 for samrai

File: tests/simulator/__init__.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def basicSimulatorArgs(dim: int, interp: int, **kwargs):
     b0 = [[3] * dim, [8] * dim]
     args = {
         "interp_order": interp,
-        "smallest_patch_size": [5] * dim,
+        "smallest_patch_size": [6] * dim,
```
Comment (nicolasaunai) [inline tests/simulator/__init__.py:29]: magic values?

File: tests/simulator/test_advance.py [hunk 4]
```diff
@@ -26,9 +27,9 @@ def ddt_test_id(self):
 
     def getHierarchy(self, interp_order, refinement_boxes, qty, nbr_part_per_cell=100,
                      diag_outputs="phare_outputs",
-                     smallest_patch_size=5, largest_patch_size=20,
+                     smallest_patch_size=6, largest_patch_size=20,
```
Comment (nicolasaunai) [inline tests/simulator/test_advance.py:30]: magic 6?

File: tests/simulator/test_initialization.py [hunk 5]
```diff
@@ -551,8 +551,8 @@ def test_no_patch_ghost_on_refined_level_case(self, simInput):
     _has_patch_ghost_on_refined_level_case = (
       {
         "cells": 40,
-        "smallest_patch_size": 5,
-        "largest_patch_size": 5},
+        "smallest_patch_size": 6,
```
Comment (nicolasaunai) [inline tests/simulator/test_initialization.py:554]: magic 6

**Review by nicolasaunai — COMMENTED** [review]
:+1:

**Comment by PhilipDeegan** [comment]
obsoleted by #551

**Pull request closed without merging**